### DISCUSSION TITLE
How to configure metrics output to file

### DISCUSSION THREAD

**Discussion by andrew-blackman** [apiman/apiman#2496] [Help]
Having downloaded and installed APIMan 3.1 we are trying to configure the metrics to be output to file.

We found a reference to a parameter write-to but cant find any further details and or examples of the required configuration.

Hope someone can help on this topic.

**Comment by msavy** [comment]
Hello again, Andrew. Please try searching the docs again for `write-to` and it should come up now. It seems the Algolia Search index was not updating for whatever reason. Simultaneously, Google has not deigned the docs worthy of fully indexing yet (no idea why).

**Reply by msavy** [reply]
Did that work for you?

**Reply by andrew-blackman** [reply]
I found a reference to the property yes.  However, is the property just write-to or apiman-manager.metrics.es.write-to or something else?

Thanks for commenting on this :) its much appreciated.

**Reply by msavy** [reply]
In this case it would be:

`apiman-gateway.metrics.write-to=LOG`

I can see the documentation is confusing on this, because examples for the manager and gateway are mixed onto the same page — most options can be used with either manager or gateway, but not this particular one (even this is a recent thing, all options were applicable to both components previously).

For the Apiman Manager application, the options are used for reading from Elasticsearch so that it can display some metrics visualisations in the UI.

I'll have a think how I can make the documentation clearer, and I will commit an improvement to the documentation of this area shortly.

Until then, here's an example snippet for you:

If you are using Vert.x. Gateway, then you simply translate this into its JSON equivalent.

```properties
# Your other Elasticsearch gateway metrics options here, if needed...

# Choose one...

# Write to log file
apiman-gateway.metrics.write-to=log

# Write to log file and server
apiman-gateway.metrics.write-to=log,remote

# Write to remote server only (the default)
apiman-gateway.metrics.write-to=remote
```

Sidebar: for Apiman 4, I'm considering unifying to just use a single configuration format for everything instead of juggling JSON and Properties for different platforms, which I think is prone to causing confusion.

Thank you for your feedback.

**Reply by andrew-blackman** [reply]
Thank you very much for you advise.  Will test this today and let you know how it goes.

**Reply by andrew-blackman** [reply]
> Thank you very much for you advise. Will test this today and let you know how it goes.

That worked perfectly.  Thank you for your assistance with this its very much appreciated.

**Reply by msavy** [reply]
Thanks for the confirmation, Andy. 

I also replied to your email 👍.
